fulp_modules/modular_bloodsuckers/bloodsucker/datum_bloodsucker.dm

```diff
@@ -0,0 +1,742 @@
+//
+//	4) GOURMOND:		Oh, to taste all the delicacies the station has to offer! DRINK ## BLOOD FROM VICTIMS WHO LIVE, EAT ## ORGANS FROM VICTIMS WHO LIVE
+
+
+//			Vassals
+//
+// - Loyal to (and In Love With) Master
+// - Master can speak to, summon, or punish his Vassals, even while asleep or torpid.
+// - Master may have as many Vassals as they want
+
+
+
+// 			Dev Notes
+//
+// HEALING: Maybe Vamps metabolize specially? Like, they slowly drip their own blood into their system?
+//			- Give Vamps their own metabolization proc, perhaps?
+//			** shadowpeople.dm has rules for healing.
+//
+// KILLING: It's almost impossible to track who someone has directly killed. But an Admin could be given
+//			an easy way to whip a Bloodsucker for cruel behavior, as a RP mechanic but not a punishment.
+//			**
+//
+// HUNGER:  Just keep adjusting mob's nutrition to Blood Hunger level. No need to cancel nutrition from eating.
+//			** mob.dm /set_nutrition()
+//			** snacks.dm / attack()  <-- Stop food from doing anything?
+
+// ORGANS:  Liver
+//			** life.dm /handle_liver()
+//
+// CORPSE:	Most of these effects likely go away when using "Masquerade" to appear alive.
+//			** status_procs.dm /adjust_bodytemperature()
+//			** traits.dm /TRAIT_NOBREATH /TRAIT_SLEEPIMMUNE /TRAIT_RESISTCOLD /TRAIT_RADIMMUNE  /TRAIT_VIRUSIMMUNE
+//			*  MASQUERADE ON/OFF: /TRAIT_FAKEDEATH (M)
+//			* /TRAIT_NIGHT_VISION
+//			* /TRAIT_DEATHCOMA <-- This basically makes you immobile. When using status_procs /fakedeath(), make sure to remove Coma unless we're in Torpor!
+//			* /TRAIT_NODEATH <--- ???
+//			** species  /NOZOMBIE
+//			* ADD: TRAIT_COLDBLOODED <-- add to carbon/life.dm /natural_bodytemperature_stabilization()
+//
+// MASQUERADE	Appear as human!
+//				** examine.dm /examine() <-- Change "blood_volume < BLOOD_VOLUME_SAFE" to a new examine
+//
+// NOSFERATU ** human.add_trait(TRAIT_DISFIGURED, "insert_vamp_datum_here") <-- Makes you UNKNOWN unless your ID says otherwise.
+// STEALTH   ** human_helpers.dm /get_visible_name()     ** shadowpeople.dm has rules for Light.
+//
+// FRENZY	** living.dm /update_mobility() (USED TO be update_canmove)
+//
+// PREDATOR See other Vamps!
+//		    * examine.dm /examine()
+//
+// WEAKNESSES:	-Poor mood in Chapel or near Chaplain.  -Stamina damage from Bible
+
+
+
+	//message_admins("DEBUG3: attempt_cast() [name] / [user_C.handcuffed] ")
+
+
+// TODO:
+//
+// Death (fire, heart, brain, head)
+// Disable Life: BLOOD
+// Body Temp
+// Spend blood over time (more if imitating life) (none if sleeping in coffin)
+// Auto-Heal (brute to 0, fire to 99) (toxin/o2 always 0)
+//
+// Hud Icons
+// UI Blood Counter
+// Examine Name (+Masquerade, only "Dead and lifeless" if not standing?)
+//
+//
+// Turn vamps
+// Create vassals
+//
+
+
+// FIX LIST
+//
+
+
+/////////////////////////////////////
+
+		// HUD! //
+
```

Comment:
Uncomment the other shit and add this here:
```
/datum/antagonist/bloodsucker/proc/update_bloodsucker_icons_added(datum/mind/m)
	var/datum/atom_hud/antag/vamphud = GLOB.huds[ANTAG_HUD_BLOODSUCKER]
	vamphud.join_hud(owner.current)
	set_antag_hud(owner.current, "bloodsucker") // "bloodsucker"
	owner.current.hud_list[ANTAG_HUD].icon = image('icons/mob/bloodsucker_icons.dmi', owner.current, "bloodsucker")	// FULP ADDITION! Check prepare_huds in mob.dm to see why.

/datum/antagonist/bloodsucker/proc/update_bloodsucker_icons_removed(datum/mind/m)
	var/datum/atom_hud/antag/vamphud = GLOB.huds[ANTAG_HUD_BLOODSUCKER]
	vamphud.leave_hud(owner.current)
	set_antag_hud(owner.current, null)


/datum/atom_hud/antag/bloodsucker // default antag where we can override behavior below
```

Doing this I was able to get the huds (mostly) working

Reply:
I tried this and I feel like it only broke the huds more, I already added replacements to give said huds using new things /tg/ gave us